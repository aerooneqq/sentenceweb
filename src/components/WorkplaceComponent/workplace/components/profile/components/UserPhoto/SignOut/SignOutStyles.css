#signOutBtn{
  width: 150px;
  height: 40px;

  margin-top: 50px;

  background: white;

  border-width: 1px;
  border-style: solid;
  border-color: orange;
  border-radius: 5px;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

#signOutBtn:focus{
  outline: none;
}

#signOutBtn:hover{
    border-color: orangered;
}

#signOutBtn:hover #signOutBtnImgContainer{
  background: url("../../../img/signOutIconBtn.png");
}
#signOutBtn:hover #signOutBtnText{
    color: orangered;
}
#signOutBtnImgContainer{
  background: url("../../../img/signOutBtnIcon.png");
  width: 24px;
  height: 24px;
}

#signOutBtnText{
  font-family: Arial;
  font-size: 17px;
  color: orange;

  margin-left: 5px;
}
