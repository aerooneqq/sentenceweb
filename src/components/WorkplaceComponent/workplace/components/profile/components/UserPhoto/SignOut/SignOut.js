import React, {Component} from "react"

//Styles
import "./SignOutStyles.css"

//Icons
import signOutIcon from "./img/sign_out_icon.svg";

/**
 * PROPS LIST:
 * 1) signOut - the function to sign out from the system.
 */
export default class SignOut extends Component{
  constructor(){
    super();

    this.state = { 
      isHovered: false
    };

    this.handleSignOutClick = this.handleSignOutClick.bind(this);
  }

  handleSignOutClick() {
    if (this.props.signOut)
      this.props.signOut();
  }

  render() {
    return (
      <div className = "toolTipContainer">
        <button id = "signOutBtn" onClick = {this.handleSignOutClick}>
          <object type="image/svg+xml"
                  data = {signOutIcon}
                  className = "signOutIcon">
            Folder     
          </object>
        </button>

        <span className = "toolTipText">Sign out</span>
      </div>
    )
  }
}
