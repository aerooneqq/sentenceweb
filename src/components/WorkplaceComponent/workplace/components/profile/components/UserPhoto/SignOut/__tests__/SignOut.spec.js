import React from 'react'
import { shallow, render } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import SignOut from "../SignOut";

describe("SignOut component", () => { 
    it("should render correctly", () => { 
        const signOut = shallow(<SignOut />);

        expect(shallowToJson(signOut)).toMatchSnapshot();
    });

    it("should call signOut from props when button clicked", () => {
        const props = { 
            signOut: () => { }
        }

        const spy = jest.spyOn(props, "signOut").mockImplementation(() => {});

        const signOut = shallow(<SignOut {...props} />);
        signOut.find("div.toolTipContainer").find("button").simulate("click");
        expect(spy).toBeCalled();
    });

    it("should work correctly when props are undefined", () => { 
        const props = { 
            signOut: undefined
        }

        const signOut = shallow(<SignOut {...props} />);
        signOut.find("div.toolTipContainer").find("button").simulate("click");
    });

    it("should work correctly when props are null", () => { 
        const props = { 
            signOut: null
        }

        const signOut = shallow(<SignOut {...props}/>);
        signOut.find("div.toolTipContainer").find("button").simulate("click");
    });
});