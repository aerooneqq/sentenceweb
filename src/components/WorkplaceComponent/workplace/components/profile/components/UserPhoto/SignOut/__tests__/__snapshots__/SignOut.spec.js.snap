// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut component should render correctly 1`] = `
<div
  className="toolTipContainer"
>
  <button
    id="signOutBtn"
    onClick={[Function]}
  >
    <object
      className="signOutIcon"
      data="sign_out_icon.svg"
      type="image/svg+xml"
    >
      Folder
    </object>
  </button>
  <span
    className="toolTipText"
  >
    Sign out
  </span>
</div>
`;
